File: app/src/main/java/droiddevelopers254/droidconke/repository/SessionFeedbackRepo.java
package droiddevelopers254.droidconke.repository;

import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.MutableLiveData;

import com.google.firebase.firestore.FirebaseFirestore;

import droiddevelopers254.droidconke.datastates.FeedBackState;
import droiddevelopers254.droidconke.models.SessionsUserFeedback;

public class SessionFeedbackRepo {

    public SessionFeedbackRepo(){

    }

    //we using mutable live data to handle cases for both a success and failure case
    public LiveData<FeedBackState> sendFeedBack(SessionsUserFeedback sessionFeedBack){
        final MutableLiveData<FeedBackState> feedBackStateMutableLiveData= new MutableLiveData<>();
        FirebaseFirestore firebaseFirestore = FirebaseFirestore.getInstance();
        firebaseFirestore.collection("sessionsFeedback")
                .add(sessionFeedBack)
                .addOnSuccessListener(documentReference -> feedBackStateMutableLiveData.setValue(new FeedBackState("Thank you for your feedback")))
                .addOnFailureListener(error ->
                        feedBackStateMutableLiveData.setValue(new FeedBackState(error.getMessage())));

        return feedBackStateMutableLiveData;
    }
}

File: app/src/main/java/droiddevelopers254/droidconke/repository/SessionFeedbackRepo.kt
package droiddevelopers254.droidconke.repository

import com.google.firebase.firestore.FirebaseFirestore
import com.google.firebase.firestore.FirebaseFirestoreException
import droiddevelopers254.droidconke.datastates.Result
import droiddevelopers254.droidconke.models.SessionsUserFeedback
import kotlinx.coroutines.tasks.await

class SessionFeedbackRepo(private val firestore: FirebaseFirestore) {


    suspend fun sendFeedBack(userSessionFeedback: SessionsUserFeedback): Result<String> {
        return try {
            firestore.collection("sessionsFeedback")
                    .add(userSessionFeedback)
                    .await()
            Result.Success("Thank you for your feedback")
        } catch (e: FirebaseFirestoreException) {
            Result.Error(e.message)
        }
    }
}